### ISSUE TITLE
`{{ post.excerpt }}` can take an array of arguments

### ISSUE THREAD

**Issue by jarednova** [timber/timber#1817]
A few steps here ...

### Acceptance Criteria
- [x] `{{ post.preview }}` is deprecated in favor of `{{ post.excerpt }}`. 
- [x] Allow the constructor to take an array of valid arguments so that this: `{{ post.excerpt({chars : 55, force: true}) }}` becomes possible within Twig templates
- [x] Update docs to document new behavior
- [x] Update upgrade guide to document new behavior

**Comment by acobster** [comment]
@jarednova deprecating `preview` in favor of `excerpt` is just about clarifying the confusing name, correct? As opposed to moving away from the `PostPreview` class to say a `PostExcerpt` class or something...

Also, should I go ahead and remove `Post::get_preview()`?

**Comment by jarednova** [comment]
@acobster yes, this is really a naming fix. Preview is used in WP to indicate post previewing (as in, viewing something from the admin before it's published).

And yes, since `{{ post.get_preview }}` has been deprecated since 1.3.1 (> 1 year) let's go ahead and remove. While we're here, we may as well rename `PostPreview` to `PostExcerpt` — lump all the pain together.

However, we should keep `{{ post.preview }}` — deprecate yes, but not fully remove until a subsequent version

**Comment by gchtr** [comment]
This issue could possibly be solved in the same update: #1345.

**Comment by acobster** [comment]
@jarednova here's a blast from the past for ya:

> While we're here, we may as well rename PostPreview to PostExcerpt — lump all the pain together.

Should we still do this? Just checking/revalidating.

@gchtr can you summarize what the fix would be for #1345?

**Comment by gchtr** [comment]
@acobster I already summarized these outstanding changes as todos in https://github.com/timber/timber/pull/2365#issuecomment-724269263.

> @gchtr can you summarize what the fix would be for #1345?

I don’t know yet. I first want to write a test to make sure this is still an issue.

**Comment by acobster** [comment]
@gchtr oh gotcha, I missed that somehow. Thanks! Seems like we can close this one out then?

**Comment by gchtr** [comment]
Yes, let’s close this so we have this off our list.
